android/autodispose-android-archcomponents/src/androidTest/java/com/uber/autodispose/android/lifecycle/KotlinExtensionsTest.kt

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2019. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.autodispose.android.lifecycle
+
+import android.util.Log
+import androidx.lifecycle.Lifecycle
+import androidx.test.annotation.UiThreadTest
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.uber.autodispose.OutsideScopeException
+import com.uber.autodispose.android.lifecycle.test.TestLifecycleOwner
+import com.uber.autodispose.test.RecordingObserver
+import io.reactivex.subjects.PublishSubject
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class KotlinExtensionsTest {
```

Comment:
These tests are basically copied over from `AndroidLifecycleScopeProviderTest`. I wanted to verify behavior and make sure the extensions don't behave differently. I can consolidate the two if the refactor looks ok